internal/bminventory/inventory.go

```diff
@@ -147,8 +156,27 @@ func generateDummyISOImage(jobApi job.API, b *bareMetalInventory, log logrus.Fie
 	dummyId := "00000000-0000-0000-0000-000000000000"
 	jobName := fmt.Sprintf("dummyimage-%s-%s", dummyId, time.Now().Format("20060102150405"))
 	imgName := fmt.Sprintf("discovery-image-%s", dummyId)
-	if err := jobApi.Create(context.Background(), b.createImageJob(jobName, imgName, "Dummy")); err != nil {
-		log.WithError(err).Errorf("failed to generate dummy ISO image")
+	if b.Config.Target == "disconnected" {
```

Comment:
I don't think that the code at this level needs to handle this target.
I think that we should have a component that handles the generating of the image.
It should be passed to the inventory from main.
It should have two implementations, one using a job, and the other, without.

@filanov, what do you think?

Reply:
Definitely not here.
We can talk about how to change the job interface to be generic, maybe having a default configuration or something like this
This is something that need to be handled in the main, it's one of the reasons we are working with interfaces, so we will be able to use dependency injection from the main.

Reply:
@filanov @ronniel1 Agree that doing b.Config.Target is not ideal here. We have also been investigating how to make the job.API interface more generic to accommodate both k8s jobs and local execution. The version you are seeing here is just to get something running quickly as a POC. I'm hoping to have the more generic version ready soon for your review.

Reply:
I was looking at the code that use the job interface and have an idea.
This code is used in two places, `generateClusterInstallConfig` and `GenerateClusterISO` in both cases you will need to implement disconnected methods.
Maybe the cleanest way to do it is to create two interfaces 
```
type ISO interface {
	GenerateClusterISO(ctx context.Context, name, imageName, ignitionConfig string) error
}
type InstallConfigGenerator interface {
	Generate(ctx context.Context, name string, clusterID strfmt.UUID, cfg []byte) error
}
```
Then we will have two implementations, one for disconnected and one for k8s, and we will select the right one in main file. 
The other parameters like namespace and s3 credentials are part of the configuration for the specific implementation and are not configurable on the fly so there should no be a problem to implement it.
What do you think?

Reply:
@filanov @ronniel1 I've submitted a PR with that has a local job implementation of the job API. https://github.com/filanov/bm-inventory/pull/285 Let me know what you guys think. I'm open to ideas if you were thinking of taking the job API interface into a different direction.

Reply:
It's a bit strange that you need to implement several dummy functions. 
As i wrote above i think that having the job interface to be used for specific k8s implementation will be cleaner.

Reply:
So inventory.go will not need to use the job interface.

Reply:
@filanov Sorry, I missed your interface suggestion before I posted my last comment. The two interfaces looks like a good direction. I'll try implementing it and will post an update for review.